### PR TITLE
Feature: SQLite Support

### PR DESCRIPTION
**Pull request by sgammon**
Ready for review [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=elide-dev&utm_campaign=badge_info)

## Summary

Adds support for SQLite, with access available via host or guest-side methods; small build fixes otherwise. Does not ship transaction support yet.

- [x] Basic SQLite support
- [x] Guest-side API
- [x] Testsuite for SQLite
- [x] Tests for guest-side calls
- [x] Transaction support
- [x] Query renderer

## Changelog

- feat(sqlite): add native sqlite driver
- feat(js): add js module for sqlite access
- feat(base): new `BoundResource` / `LifecycleBoundResources` pattern in `base`
- feat(engine): new core native library loader, intended to complement static JNI
- chore: update runtime modules
- chore: update to latest sqlite jdbc/jni version

### Known Issues

We will need to patch the JS module loader to support our custom built-ins prefix, i.e. `import x from 'elide:something';`. In the meantime, built-in Elide modules are exposed at `<module>`, with no prefix.

### CODE REVIEW

File: packages/graalvm/src/main/kotlin/elide/runtime/gvm/internals/sqlite/SqliteQueryRenderer.kt [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+  // any helpful advice to fix the error.
+  private fun queryRenderError(
+    subject: String,
+    ctx: QueryTemplateContext? = null,
+    message: String,
+    position: Int = -1,
+    soFar: Int = -1,
+  ): Throwable {
+    TODO("no error rendering yet (message: $message)")
+  }
```
Comment (sgammon) [inline packages/graalvm/src/main/kotlin/elide/runtime/gvm/internals/sqlite/SqliteQueryRenderer.kt:90]: Query render errors are not implemented yet

File: packages/graalvm/src/main/kotlin/elide/runtime/gvm/sqlite/SQLite.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+import java.nio.file.Path
+import elide.runtime.gvm.internals.sqlite.SQLiteCreateDatabaseOptions
+import elide.runtime.gvm.internals.sqlite.SqliteDatabaseProxy
+import elide.runtime.intrinsics.sqlite.SQLiteDatabase
+
+/**
+ *
+ */
+public data object SQLite {
```
Comment (sgammon) [inline packages/graalvm/src/main/kotlin/elide/runtime/gvm/sqlite/SQLite.kt:23]: nit: doc

File: packages/graalvm/src/main/kotlin/elide/runtime/intrinsics/sqlite/SQLiteTransaction.kt [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+ * @see SQLiteTransactor Transactor Interface
+ */
+@API @ReflectiveAccess public interface SQLiteTransaction<R>: ProxyExecutable, ProxyObject, Future<R> {
+  /** Indicate the type of this transaction; see [SQLiteTransactionType] for options. */
+  public val type: SQLiteTransactionType
+
+  /**
+   * ## Transaction Mode: Deferred
+   */
+  @Polyglot public fun deferred(): SQLiteTransaction<R>
```
Comment (sgammon) [inline packages/graalvm/src/main/kotlin/elide/runtime/intrinsics/sqlite/SQLiteTransaction.kt:76]: nit: doc

File: packages/graalvm/src/test/kotlin/elide/runtime/gvm/js/node/GenericJsModuleTest.kt [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import java.util.stream.Stream
+import kotlin.streams.asStream
+import kotlin.test.Test
+import kotlin.test.assertContains
+import elide.runtime.intrinsics.GuestIntrinsic
+
+/**
+ *
+ */
+internal abstract class GenericJsModuleTest<T> : AbstractJsModuleTest<T>() where T: GuestIntrinsic {
```
Comment (sgammon) [inline packages/graalvm/src/test/kotlin/elide/runtime/gvm/js/node/GenericJsModuleTest.kt:27]: nit: drop

**Review by darvld — APPROVED** [review]

**Pull request merged**